File: Data Structures & Algorithms/max-water-container/submission-1.java
class Solution {
    public int maxArea(int[] heights) {
        int left = 0;
        int right = heights.length - 1;
        int max = 0;
        while (left < right) {
            int l = heights[left];
            int r = heights[right];
            int curr = Math.min(l, r) * (right - left);
            max = Math.max(curr, max);
            if (l < r) {
                left++;
            } else {
                right--;
            }
        }
        return max;
    }
}

File: Data Structures & Algorithms/trapping-rain-water/submission-1.py
class Solution:
    def trap(self, height: List[int]) -> int:
        maxLeft = 0
        maxRight = 0
        left = 0
        right = len(height) - 1
        total = 0
        while (left < right):
            currLeft = height[left]
            currRight = height[right]
            maxLeft = max(maxLeft, currLeft)
            maxRight = max(maxRight, currRight)
            if maxLeft < maxRight:
                total += maxLeft - currLeft
                left+=1
            else:
                total += maxRight - currRight
                right-=1
        return total
            

        